datasets/eg/terrorists/crawler.py

```diff
@@ -0,0 +1,168 @@
+
+
+def parse_date(date_str: str) -> datetime:
+
+    if not date_str:
+        return None
+
+    western_date = arabic_to_western(date_str)
+
+    # We first check if it's just one date
+    try:
+        return datetime.strptime(western_date, "%Y/%m/%d")
+    except ValueError:
+        # Otherwise we check if the first line is a date
+        try:
+            return datetime.strptime(western_date.split("\n")[0], "%d/%m/%Y")
+        except ValueError:
+            return None
+
+
+def parse_sheet(
+    sheet: openpyxl.worksheet.worksheet.Worksheet,
+    skiprows: int = 0,
+) -> Generator[dict, None, None]:
+    headers = None
+
+    row_counter = 0
+
+    for row in sheet.iter_rows():
+        # Increment row counter
+        row_counter += 1
+
+        # Skip the desired number of rows
+        if row_counter <= skiprows:
+            continue
+        cells = [c.value for c in row]
+        if headers is None:
+            headers = []
+            for idx, cell in enumerate(cells):
+                if cell is None:
+                    cell = f"column_{idx}"
+                headers.append(slugify(cell, "_").lower())
+            continue
+
+        record = {}
+        for header, value in zip(headers, cells):
+            if isinstance(value, datetime):
+                value = value.date()
+            record[header] = stringify(value)
+        if len(record) == 0:
+            continue
+        yield record
+
+
+def crawl_terrorist(input_dict: dict, context: Context):
+
+    first_name = input_dict.pop("alasm")
+    last_name = input_dict.pop("asm_akhr")
+
+    person = context.make("Person")
+    person.id = context.make_id(first_name, last_name)
+    h.apply_name(person, first_name=first_name, last_name=last_name)
+    person.add("nationality", input_dict.pop("aljnsyt"), lang="ar")
+    person.add("passportNumber", input_dict.pop("jwaz_alsfr"))
+    person.add("idNumber", input_dict.pop("alrqm_alqwmy"))
+    person.add("topics", "crime.terror")
+
+    sanction = h.make_sanction(context, person)
+
+    sanction.add("listingDate", parse_date(input_dict.pop("tarykh_alnshr")))
+    sanction.add("recordId", input_dict.pop("rqm_alqdyt"))
+    sanction.add(
+        "summary",
+        "Terrorist Listing Decision Number: {}, Publication Page: {}".format(
+            input_dict.pop("rqm_qrar_adraj_alarhabyyn"), input_dict.pop("dd_alnshr")
+        ),
+    )
+
+    context.emit(person, target=True)
+    context.emit(sanction)
+    context.audit_data(input_dict)
+
+
+def crawl_terrorist_entities(input_dict: dict, context: Context):
+
+    name = input_dict.pop("asm_alkyan")
+
+    if not name:
+        return
+
+    entity = context.make("LegalEntity")
+    entity.id = context.make_id(name)
+    entity.add("name", name)
+
+    sanction = h.make_sanction(context, entity)
+    sanction.add("recordId", input_dict.pop("rqm_alqdyt"))
+    sanction.add("summary", input_dict.pop("althdythat"), lang="ar")
```

Comment:
could you make all the langs 3-letter?

```suggestion
    sanction.add("summary", input_dict.pop("althdythat"), lang="ara")
```